ecl/hql/hqlopt.cpp

```diff
@@ -30,6 +30,22 @@
 
 #define MIGRATE_JOIN_CONDITIONS             // This works, but I doubt it is generally worth the effort. - maybe on a flag.
 //#define TRACE_USAGE
+#ifdef _DEBUG
+//#define TRACK_EXPR_SEQ  nnnnn
+#endif
+//#define CHECK_USAGE_COUNT                 // For tracking down issues where usage count is inconsistent (may fail for datasets child queries)
+
+#ifdef TRACK_EXPR_SEQ
+void CHECK_EXPR_ID(IHqlExpression * expr)
+{
+    if (querySeqId(expr->queryBody()) == TRACK_EXPR_SEQ)
+        expr->numChildren();
```

Comment:
Don't understand how getting numChildren()  has any effect?  Wouldn't this be optimised  away?